native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ArgumentHandler.java

```diff
@@ -161,6 +196,111 @@ private Object getArgumentValue(BObject argumentNode, Type parameterType) {
         }
     }
 
+    private Object getIdArgumentValue(BObject argumentNode, Type parameterType) {
+        if (argumentNode.get(VALUE_FIELD) instanceof BString) {
+            String obj = ((BString) argumentNode.get(VALUE_FIELD)).getValue();
+            if (parameterType.getTag() == TypeTags.STRING_TAG) {
+                return StringUtils.fromString(obj);
+            } else if (parameterType.getTag() == TypeTags.INT_TAG) {
+                return Integer.parseInt(obj);
+            } else if (parameterType.getTag() == TypeTags.FLOAT_TAG) {
+                return ValueUtils.convert(JsonUtils.parse(obj), parameterType);
+            } else if (parameterType.getTag() == TypeTags.DECIMAL_TAG) {
+                return ValueUtils.convert(JsonUtils.parse(obj), parameterType);
+            } else if (parameterType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG) {
+                // not validating if this is uuid:Uuid since compiler plugin does that
+                return ValueCreator.createRecordValue(parameterType.getPackage(), parameterType.getName(),
+                        (BMap<BString, Object>) JsonUtils.parse(obj.replaceAll("\\\\", "")));
+            } else if (parameterType.getTag() == TypeTags.ARRAY_TAG) {
+                return getArrayTypeIdValue(parameterType, obj);
+            } else if (parameterType.getTag() == TypeTags.UNION_TAG) {
+                return getUnionTypeIdValue(parameterType, argumentNode);
+            }
+        }
+        return argumentNode.get(VALUE_FIELD);
+    }
+
+    private Object getUnionTypeIdValue(Type parameterType, BObject argumentNode) {
+        List<Type> members = ((UnionType) parameterType).getMemberTypes();
+        for (Type member : members) {
+            if (member.getTag() != TypeTags.NULL_TAG && member.getTag() != TypeTags.ERROR_TAG) {
+                return getIdArgumentValue(argumentNode, member);
+            }
+        }
+        return null;
+    }
+
+    private Object getArrayTypeIdValue(Type parameterType, String obj) {
+        Type memberType = ((ArrayType) parameterType).getElementType();
+        String[] stringArray = obj.replaceAll("\\[", "")
+                .replace("]", "").replace(" ", "")
+                .split(",");
```

Comment:
We should get a string array here. No need to do string manipulation to get the array of ids.

Reply:
Only string and int are allowed to pass as ID type, if it is an array it has to be passed as a string
eh: int[]
"[2, 3, 4]"

Reply:
<img width="538" alt="Screenshot 2023-06-15 at 13 09 42" src="https://github.com/ballerina-platform/module-ballerina-graphql/assets/39673615/f5515a4d-cd2d-4ca6-bcbd-0caa3e777906">

Reply:
<img width="1174" alt="Screenshot 2023-06-15 at 13 10 59" src="https://github.com/ballerina-platform/module-ballerina-graphql/assets/39673615/360be3a1-82ee-4e16-ac24-16e54008e1b0">

Reply:
string array
<img width="1174" alt="Screenshot 2023-06-15 at 13 10 59" src="https://github.com/ballerina-platform/module-ballerina-graphql/assets/39673615/92e56281-531d-4667-b2b6-7b917bce8ee0">